File: DataModel/Models/Compte.cs
using System;
using System.Collections.Generic;

namespace DataModel.Models
{
    public partial class Compte
    {
        public Compte()
        {
            EcritureComptable = new HashSet<EcritureComptable>();
        }

        public long IdCpt { get; set; }
        public long CodeCpt { get; set; }
        public string LibelleCpt { get; set; }
        public double? MontantCpt { get; set; }
        public int? LevelCpt { get; set; }
        public string TypeCpt { get; set; }
        public long? IdCptParent { get; set; }
        public bool? VisibilityCpt { get; set; }

        public virtual ICollection<EcritureComptable> EcritureComptable { get; set; }
        public virtual Tiers Tiers { get; set; }
        public virtual Compte IdCptParentNavigation { get; set; }
        public virtual ICollection<Compte> InverseIdCptParentNavigation { get; set; }
    }
}

File: WebProject/wwwroot/app/Domain/Models/Compte.ts
module app.Domain.Models {
    export interface ICompte {
        idCpt: number;
        codeCpt: number;
        libelleCpt: string;
        idCptParent: number;
        visibilityCpt: boolean;
        montantCpt?: number;
    }

    export class Compte extends app.Domain.GenericModel.EntityBase implements ICompte {
        constructor(public idCpt: number, public codeCpt: number, public libelleCpt: string, public idCptParent: number, public visibilityCpt: boolean, public montantCpt?: number) {

            super();
            this.idCpt = idCpt;
            this.codeCpt = codeCpt;
            this.libelleCpt = libelleCpt;
            this.idCptParent = idCptParent;
            this.visibilityCpt = visibilityCpt;
            this.montantCpt = montantCpt;
        }
    }
}